3.Tests/UI/StepDefinitions/CommonStepDefinition.cs

```diff
@@ -19,4 +36,39 @@ public void Login()
         LoginPage loginPage = new LoginPage();
         loginPage!.LoginIntoApplication();
     }
+
+    [When(@"(?:the user )?clicks on '([a-zA-Z ]+)'(?: on '([a-zA-Z ]+)')?$")]
+    public void Click(string elementName, string viewName)
+    {
+        if (viewName != null)
+        {
+            CurrentView = viewName;
+        }
+
+        UIElementFactory.GetElement(elementName, CurrentView).Click();
+    }
+
+    [When(@"(?:the user )?types ""(.*)"" on '([a-zA-Z ]+)'(?: on '([a-zA-Z ]+)')?$")]
+    public void Type(string input, string elementName, string viewName)
+    {
+        if (viewName != null)
+        {
+            CurrentView = viewName;
+        }
+
+        UIElementFactory.GetElement(elementName, CurrentView).Type(input);
+    }
+
+    [Then(@"the '(.*)' should be (not )?displayed")]
```

Comment:
"not" has to go before "be" as in: the button should **not** be displayed

Reply:
Oh you're right, my bad. Fixed